### PR TITLE
Added drawer component

### PR DESCRIPTION
**Pull request by amityt**
Signed-off-by: Amit Kumar Das <amit@chaosnative.com>

This PR adds:
- Litmus UI Drawer component

### CODE REVIEW

File: src/core/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <MuiDrawer
+      anchor={anchor}
+      open={open}
+      onClose={onClose}
+      className={`${classes.root} ${className}`}
+      {...rest}
+    >
+      <div className={classes.content}>
+        {modalActions && (
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]: we can use close icon as default

File: src/core/Drawer/Drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  onClose,
+  className,
+  modalActions,
+  ...rest
+}) => {
+  const classes = useStyles({ anchor });
+  return (
+    <MuiDrawer
+      anchor={anchor}
+      open={open}
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]: `open` can be omitted as it'll pass through `rest`

File: src/core/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+interface DrawerProps extends DrawerBaseProps {
+  onClose: () => void;
+  modalActions?: React.ReactNode;
+}
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  open,
+  anchor,
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]:
```suggestion
  anchor = "right",
```

File: src/core/Drawer/styles.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    width: (props: StyleProps) =>
+      props.anchor === "top" || props.anchor === "bottom"
+        ? "100%"
+        : "fit-content",
+    color: theme.palette.text.primary,
+    background: theme.palette.background.paper,
+    textAlign: "center",
+  },
+
+  modalActions: {},
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/styles.ts]: this is empty

File: src/core/Drawer/Drawer.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    <MuiDrawer
+      anchor={anchor}
+      open={open}
+      onClose={onClose}
+      className={`${classes.root} ${className}`}
+      {...rest}
+    >
+      <div className={classes.content}>
+        {modalActions && (
+          <div className={classes.modalActions}>{modalActions}</div>
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]:
```suggestion
          modalActions
```

File: src/core/Drawer/Drawer.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { Drawer as MuiDrawer } from "@material-ui/core";
+import React from "react";
+import { DrawerBaseProps } from "./base";
+import { useStyles } from "./styles";
+
+interface DrawerProps extends DrawerBaseProps {
+  onClose: () => void;
+  modalActions?: React.ReactNode;
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]:
```suggestion
  modalActions?: React.ReactElement;
```

File: src/core/Drawer/__tests__/Drawer.test.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { render } from "@testing-library/react";
+import React from "react";
+import { LitmusThemeProvider } from "../../../theme";
+import { Drawer } from "../Drawer";
+
+describe("Drawer Component", () => {
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/__tests__/Drawer.test.tsx]: testing can be improved refer `EditableText` component

File: src/core/Drawer/Drawer.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import { Drawer as MuiDrawer } from "@material-ui/core";
+import React from "react";
+import { DrawerBaseProps } from "./base";
+import { useStyles } from "./styles";
+
+interface DrawerProps extends DrawerBaseProps {
+  onClose: () => void;
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]: we can remove this

File: src/core/Drawer/base.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+import { DrawerProps } from "@material-ui/core";
+
+export type DrawerBaseProps = DrawerProps;
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/base.ts]:
```suggestion
export type DrawerBaseProps = Omit<DrawerProps, "transitionDuration" |  "variant" | "PaperProps" | "SlideProps" | "ModalProps">;
```

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

File: src/core/Drawer/Drawer.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+interface DrawerProps extends DrawerBaseProps {
+  onCloseButton: () => void;
+}
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  anchor = "right",
+  className,
+  onCloseButton,
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]:
```suggestion
  onButtonClose,
```

File: src/core/Drawer/Drawer.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+      {...rest}
+    >
+      <div className={classes.content}>
+        <IconButton
+          data-testid="close-btn"
+          className={classes.closeIcon}
+          onClick={onCloseButton}
+          component="span"
+        >
+          <Icon name="close" />
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/Drawer.tsx]: default is close...other icon names can be passed

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

File: src/core/Drawer/styles.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+      props.anchor === "top" || props.anchor === "bottom"
+        ? "100%"
+        : "fit-content",
+    color: theme.palette.text.primary,
+    background: theme.palette.background.paper,
+    textAlign: "center",
+  },
+  closeIcon: {
+    float: "right",
+    width: "3rem",
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/styles.ts:25]:
```suggestion
    width: "fit-content",
```

File: src/core/Drawer/styles.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    height: "3rem",
```
Comment (arkajyotiMukherjee) [inline src/core/Drawer/styles.ts:26]: either remove or auto

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

File: src/core/Drawer/styles.ts (hunk 12 above)
Comment (amityt) [reply src/core/Drawer/styles.ts:25]: Tried this change, the iconbutton was still in oval shape.
Screenshot from 2021-08-25 12-09-39

**Review by arkajyotiMukherjee — APPROVED** [review]

**Pull request merged**